sound/soc/sof/control.c

```diff
@@ -489,3 +512,25 @@ int snd_sof_bytes_ext_get(struct snd_kcontrol *kcontrol,
 
 	return 0;
 }
+
+int snd_sof_control_type(type)
+{
+	int ret;
+
+	switch (type) {
+	case SOF_TPLG_KCTL_VOL_ID:
+	case SOF_TPLG_KCTL_SWITCH_ID:
+		ret = SND_SOC_TPLG_TYPE_MIXER;
+		break;
+	case SOF_TPLG_KCTL_BYTES_ID:
+		ret = SND_SOC_TPLG_TYPE_BYTES;
+		break;
+	case SOF_TPLG_KCTL_ENUM_ID:
+		ret = SND_SOC_TPLG_TYPE_ENUM;
+		break;
+	default:
+		ret = -EINVAL;
```

Comment:
how many lines of code you'd save, if you just did `return SND_SOC_TPLG_TYPE_x;` and removed all the `break`s and the `ret` variable ;-)

Reply:
yeah, sure can do :) just have some vague remembrance in my head for someone complaining about these "excessive" returns... also not sure if needed to add some other logic later, this would be then better.  But it can be changed then.